### PR TITLE
optimize: prewire task dependencies on orchestrator side

### PR DESCRIPTION
**Pull request by Crane-Liu**
## Summary

Move dependency wiring to the orchestrator side for the a2a3 and a5 `tensormap_and_ringbuffer` runtimes, and remove the old scheduler-side deferred wiring fallback.

The latest cleanup makes this PR a single Orch-side wiring design instead of a mixed O/S fallback path:

- Zero-fanin tasks are published ready directly by Orch.
- Fanin tasks whose producers are already complete are published ready directly by Orch.
- Fanin tasks with live producers are wired by Orch into the dependency pool before scheduling.
- Scheduler threads no longer drain a deferred wiring queue.
- S0/S1/S2 can still complete dependency-only/dummy fanout in parallel.
- a2a3 and a5 now follow the same converged structure.

### a2a3 runtime

- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/pto_orchestrator.cpp`
  - Move no-fanin, completed-fanin, and live-fanin wiring/publish decisions into Orch submit.
  - Remove fallback branches that sent dependency wiring back to the scheduler path.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/scheduler/pto_scheduler.h`
  - Remove old SPSC wiring queue helpers and deferred wiring drain logic.
  - Keep the shared fanin wiring helper used by Orch-side live-fanin wiring.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/scheduler/scheduler_dispatch.cpp`
  - Remove the scheduler dispatch phase that drained deferred wiring work.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/scheduler/scheduler_cold_path.cpp`
  - Align cold-path completion/fanout handling with the new Orch-wired dependency state.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/shared/pto_runtime2_init.cpp`
  - Remove wiring queue layout/init/reset/destroy plumbing.
  - Update runtime initialization around Orch-owned dependency wiring state.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/pto_runtime2.{cpp,h}`
  - Remove stale wiring queue APIs and state exposure.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/pto_runtime2_types.h`
  - Remove deprecated wiring state fields tied to the fallback queue.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/pto_ring_buffer.h`
  - Update ring/dependency-pool comments and ownership assumptions.
- `src/a2a3/runtime/tensormap_and_ringbuffer/runtime/scheduler/{scheduler_context.h,scheduler_types.h}`
  - Remove unused scheduler-side wiring/drain fields.

### a5 runtime

- Applied the same convergence to the matching a5 files under `src/a5/runtime/tensormap_and_ringbuffer/runtime/`.
- a5 now uses the same Orch-side submit/wiring model as a2a3 instead of carrying an independent fallback path.

### Shared profiling / tooling

- `src/a2a3/platform/include/common/l2_swimlane_profiling.h`
- `src/a5/platform/include/common/l2_swimlane_profiling.h`
- `simpler_setup/tools/swimlane_converter.py`

These were updated so the old scheduler-wire swimlane/profiling label is treated as legacy scheduler wire phase metadata rather than an active deferred wiring phase.

## Documentation Updates

Updated both a2a3 and a5 runtime docs:

- `src/a2a3/runtime/tensormap_and_ringbuffer/docs/RUNTIME_LOGIC.md`
- `src/a2a3/runtime/tensormap_and_ringbuffer/docs/MULTI_RING.md`
- `src/a2a3/runtime/tensormap_and_ringbuffer/docs/SUBMIT_BY_CLUSTER.md`
- `src/a5/runtime/tensormap_and_ringbuffer/docs/RUNTIME_LOGIC.md`
- `src/a5/runtime/tensormap_and_ringbuffer/docs/MULTI_RING.md`
- `src/a5/runtime/tensormap_and_ringbuffer/docs/SUBMIT_BY_CLUSTER.md`

The docs now describe:

- Orch-side dependency wiring as the primary path.
- Removal of the scheduler-side deferred wiring queue fallback.
- Updated ownership of dependency-pool writes.
- The remaining scheduler responsibility: execute ready work and release fanout.
- The fact that S0/S1/S2 can complete dependency-only/dummy fanout in parallel.

## Test Updates

- `tests/ut/cpp/a2a3/test_wiring.cpp`
- `tests/ut/cpp/a5/test_wiring.cpp`

Updated wiring tests to exercise Orch-side helpers directly.

Removed obsolete tests for the deleted SPSC wiring queue:

- `tests/ut/cpp/a2a3/test_spsc_queue.cpp`
- `tests/ut/cpp/a5/test_spsc_queue.cpp`

Updated:

- `tests/ut/cpp/CMakeLists.txt`

## Current Behavior

1. `wfanin == 0`: Orch seeds fanin state, records the dep-pool position, and routes the task ready.
2. `wfanin > 0` and all producers are already complete: Orch seeds fanin/dispatch state and routes the task ready.
3. `wfanin > 0` with unfinished producers: Orch wires fanout entries through the shared wiring helper before the task becomes visible to schedulers.
4. Scheduler threads complete ready tasks and release fanout; they do not perform deferred wiring drain anymore.

### Previous 100-round Strategy Sweep

Benchmark: `qwen3_14b_decode`, `StressBatch16Seq3500`, `a2a3`, device 4, `tensormap_and_ringbuffer`, 100 rounds per run. Values below are trimmed averages, dropping 10 low and 10 high rounds.

Negative delta means faster.

| Set | Device us | Device vs Baseline | Sched us | Sched vs Baseline | Orch us |
| --- | ---: | ---: | ---: | ---: | ---: |
| Baseline (`simpler-base`) | 2436.5 | +0.0 | 2105.2 | +0.0 | 383.3 |
| S0/S1/S2 DummyDrain avg | 2416.9 | -19.6 | 2091.5 | -13.7 | - |
| O-side prewire avg | 2400.4 | -36.1 | 2075.2 | -30.0 | 422.1 |

Readout:

- Device improves by 36.1 us vs baseline, about 1.48%.
- Scheduler completion improves by 30.0 us vs baseline, about 1.43%.
- Compared with the S0/S1/S2 dummy-drain-only version, O-side prewire adds another 16.5 us Device improvement and 16.3 us Sched improvement.
- Orch time increases by about 38.8 us vs baseline, which is the expected work shift from S to O. For qwen, O-side completion still has enough slack that the total Device/Sched time improves.

### Latest pypto-lib qwen14b Decode 40L Check

Command:

```bash
task-submit --device 0 --max-time 0 --timeout 0 --run 'cd /data/pyptouser/zhangtao/zt/pypto-lb && .venv-bench/bin/python models/qwen3/14b/decode_layer.py -p a2a3 -d $TASK_DEVICE --validate-fwd --fwd-layers 40'
```

Result from `after_lock_cleanup_qwen14b_40l_20260703_162637_dev0.log`:

| Metric | Time |
| --- | ---: |
| `device_wall` | 35.527200 ms |
| `orch` | 31.077500 ms |
| `sched` | 35.157720 ms |
| `runner_run` | 37.523351 ms |

Correctness:

- `argmax match 16/16`
- `sample match 16/16`
- `logits 100.0000% within 5e-2`
- `max_abs_err=0.0199`

Compared with the earlier device-0 baseline average for this 40L check, `device_wall` and `sched` are within about 0.5%, so the cleanup does not show an obvious performance regression.

## Tests

- `git diff --check`
- `cmake --build tests/ut/cpp/build -j$(nproc)`
- `ctest --test-dir tests/ut/cpp/build --output-on-failure -R '^(test_wiring|test_a2a3_orchestrator_fanin|test_a5_wiring|test_a5_orchestrator_fanin)$'`
- pypto-lib qwen14b decode 40L validation:
  - `models/qwen3/14b/decode_layer.py -p a2a3 -d $TASK_DEVICE --validate-fwd --fwd-layers 40`

### CODE REVIEW

File: src/a2a3/runtime/tensormap_and_ringbuffer/runtime/pto_orchestrator.cpp [hunk 1]
```diff
@@ -864,12 +903,33 @@ static TaskOutputTensors submit_task_common(
...
+    } else if (all_claimed_fanin_completed(fanin_builder)) {
+        int32_t ready_seed = fanin_builder.count + 1;
+        cur_slot_state.fanin_count = ready_seed;
+        payload.dispatch_fanin.store(fanin_builder.count, std::memory_order_release);
+        cur_slot_state.fanin_refcount.store(ready_seed, std::memory_order_release);
+        (void)route_orch_inline_ready(sched, cur_slot_state);
+        inline_wired = true;
+    } else if (try_orch_prewire_task(sched, cur_slot_state, fanin_builder.count)) {
+        inline_wired = true;
+    }
```
[the remark this answers is not in the document]
Comment (ChaoWao) [reply src/a2a3/runtime/tensormap_and_ringbuffer/runtime/pto_orchestrator.cpp]: Addressed in `3b90ab6c` (*refactor: simplify orchestrator dependency wiring*). Both inline-ready fast paths (zero-fanin and all-producers-completed) now call `mark_orch_wired_no_edges` → `mark_dep_pool_position`, which sets `dep_pool_mark = dep_pool.top` — the same watermark `wire_task` establishes — so `PTO2DepListPool::reclaim` keeps advancing the tail correctly when an inline task becomes the live-window head. (This comment was against the earlier `route_orch_inline_ready` / `try_orch_prewire_task` naming, pre-`3b90ab6c`.) No further change needed; resolving as stale.

File: src/a2a3/runtime/tensormap_and_ringbuffer/runtime/scheduler/scheduler_dispatch.cpp [hunk 2]
```diff
@@ -1056,44 +1056,15 @@ int32_t SchedulerContext::resolve_and_dispatch(Runtime *runtime, int32_t thread_
...
         // Dependency-only tasks bypass AICore dispatch: they go through the
         // scheduler so fanin/fanout edges stay consistent, but completion is
-        // signalled inline here. Pinned to thread 0 to avoid cross-thread
-        // races and to keep cache hot near the wiring drain above.
-        if (thread_idx == 0) {
-            constexpr int DUMMY_DRAIN_BATCH = 16;
+        // signalled inline here. The ready queue is MPMC, and the fanout path
+        // uses per-slot locks/atomics, so multiple scheduler threads can share
+        // the dependency-only resolve work.
+        if (thread_idx < 3) {
```
Comment (ChaoZheng109) [inline src/a2a3/runtime/tensormap_and_ringbuffer/runtime/scheduler/scheduler_dispatch.cpp:1066]: 进入调度逻辑的线程一定是小于3的，这里是多余的判断
